### DISCUSSION TITLE
[rust-analyzer] Workspace symbol picker should search the same symbol types as the single-file symbol picker

### DISCUSSION THREAD

**Discussion by willhansen** [helix-editor/helix#8256] [General]
# The Problem

With rust, the workspace symbol picker and the single-file symbol picker have different search behavior, even with a single file in the project.  This is unintuitive, and the default behavior should be that they are the same.

- The workspace symbol picker does not return any function names unless there are absolutely no struct names that fuzzy match the given string.
- The single-file symbol picker searches both from the beginning

# Reproduction:
 
With a file generated by this python script:
```
#!/usr/bin/env python3

n = 300
with open("main.rs", "w") as file:
    file.write('fn main() {println!("asdf");}\n')
    for i in range(n):
        file.write('fn a' + str(i) + '_snake_func() {println!("asdf");}\n')
        file.write('fn a' + str(i) + 'CamelFunc() {println!("asdf");}\n')
        file.write('struct a' + str(i) + '_snake_struct();\n')
        file.write('struct a' + str(i) + 'CamelStruct();\n')
```
Workspace symbol picker:
![Selection_265](https://github.com/helix-editor/helix/assets/2918280/d8f44707-cad2-4b9f-abd1-1237c324e62d)

Single-file symbol picker:
![Selection_264](https://github.com/helix-editor/helix/assets/2918280/84b71d6c-cf09-4fe5-95f2-6c6f1ea9fb69)

# The Fix

This can by fixed with this `language.toml` file, making both symbol pickers include functions from the beginning:
```
[[language]]
name = "rust"

[language.config.workspace.symbol.search]
kind = "all_symbols"
```

This should be set by default.

(continued from discussion: https://github.com/helix-editor/helix/discussions/7867)

**Comment by pascalkuthe** [comment]
There seem to be some performance concerns about changing the default https://github.com/rust-lang/rust-analyzer/issues/14553#issuecomment-1506837815.

You can also append `#` to search for functions too. It doesn't currently work in helix but #7265 will fix that. I think we should stick with the RA default. Whether a different default is the better choice or not should be discussed and changed upstream instead of here.

In the meantime adding this t your personal config is a very simple workaround
